pkg/generate/code/resource_reference_test.go

```diff
@@ -146,15 +187,33 @@ func Test_ReferenceFieldsPresent_SliceOfReferences(t *testing.T) {
 			GeneratorConfigFile: "generator-with-reference.yaml",
 		})
 
-	//NOTE: For the moment, we are substituting SecurityGroupId with ApiId
-	// just to test code generation for slices of reference
 	crd := testutil.GetCRDByName(t, g, "VpcLink")
 	require.NotNil(crd)
-	expected := "return false || (ko.Spec.SecurityGroupRefs != nil) || (ko.Spec.SubnetRefs != nil)"
-	assert.Equal(expected, code.ReferenceFieldsPresent(crd, "ko"))
+	expected :=
+		`	for _, f0iter := range ko.Spec.SecurityGroupRefs {
+		if f0iter != nil && f0iter.From != nil {
+			hasReferences = true
+			arr := f0iter.From
+			if arr.Name == nil || *arr.Name == "" {
+				return hasReferences, fmt.Errorf("provided resource reference is nil or empty: SecurityGroupRefs")
+			}
+			obj := &ec2apitypes.SecurityGroup{}
+			if err := getReferencedResourceState_SecurityGroup(ctx, apiReader, obj, *arr.Name, namespace); err != nil {
+				return hasReferences, err
+			}
+			if ko.Spec.SecurityGroupIDs == nil {
+				ko.Spec.SecurityGroupIDs = make([]*string, 0, 1)
+			}
+			ko.Spec.SecurityGroupIDs = append(ko.Spec.SecurityGroupIDs, (*string)(obj.Status.ID))
```

Comment:
The more I think about this the more i believe that maybe we should not allow "normal" and "referenced" fields at the same time (`SecurityGroupIDs` and `SecurityGroupRefs`). Using both can lead us into an area of problems that is very hard to solve (unless if we want to leverage annotations, to keep track of what was referenced and was was not). Or even introduce a new Status field to help doing so (like Conditions and ACKResourceMetadata)

Reply:
We don't allow a mix of concrete and refs. That's the point of the [`validateReferenceFields` method](https://gist.github.com/RedbackThomson/8e11cbbe96065a4eb812c387665c747d#file-references-go-L116).